### PR TITLE
Correctness

### PR DESCRIPTION
**Pull request by EECOLOR**

### CODE REVIEW

File: src/main/scala/jio/package.scala [hunk 1]
```diff
@@ -50,7 +51,11 @@ package object jio extends JioFiles {
       Files.setLastModifiedTime(path, jnfa.FileTime.from(nanoSeconds, TimeUnit.NANOSECONDS))
 
     def tryLock():jnc.FileLock     = withWriteChannel(_.tryLock)
-    def truncate(size: Long): Unit = withWriteChannel(_ truncate size )
+    def truncate(size: Long): Unit = withWriteChannel {
+      case c if c.size > size => c truncate size
+      case c if c.size < size => c write (ByteBuffer wrap Array.fill((size - c.size).toInt)(0.toByte), c.size)
+      case _                  => // sizes are equal
```
Comment (paulp) [inline src/main/scala/jio/package.scala:60]: Pretty ugly expression.

Comment (EECOLOR) [inline src/main/scala/jio/package.scala:60]: what would you suggest?

Comment (paulp) [inline src/main/scala/jio/package.scala:60]: It's not obvious what's happening.

Comment (EECOLOR) [inline src/main/scala/jio/package.scala:60]: agreed

I need to fix some other behavior here, so I will take that into account

Comment (paulp) [inline src/main/scala/jio/package.scala:60]: And having a method called "truncate" which expands the file arbitrarily large... but I guess if we're really going for the identity filesystem those are the required semantics.

File: src/main/scala/jio/package.scala [hunk 2]
```diff
@@ -51,7 +53,12 @@ package object jio extends JioFiles with DecorateAsScala with DecorateAsJava {
       Files.setLastModifiedTime(path, jnfa.FileTime.from(nanoSeconds, TimeUnit.NANOSECONDS))
 
     def tryLock():jnc.FileLock     = withWriteChannel(_.tryLock)
-    def truncate(size: Long): Unit = withWriteChannel(_ truncate size )
+    def truncate(size: Long): Unit = withWriteChannel {
+      case c if c.size > size => c truncate size
+      case c if c.size < size => c appendNullBytes (at = c.size, amount = (size - c.size).toInt)
```
Comment (EECOLOR) [inline src/main/scala/jio/package.scala:58]: is this better?

File: src/main/scala/jio/package.scala (hunk 1 above)
Comment (EECOLOR) [inline src/main/scala/jio/package.scala:60]: Yeah... I didn't like that one either. Kinda defeats the word `truncate`

File: src/main/scala/jio/package.scala (hunk 2 above)
Comment (paulp) [inline src/main/scala/jio/package.scala:58]: Yeah that's a lot better.

**Comment by paulp** [comment]
Bueno.

**Pull request merged**